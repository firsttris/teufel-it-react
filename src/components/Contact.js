import React, { Component } from 'react';

class Contact extends Component {
  render(props) {
    const margin10 = { margin: '10px' };
    return (
      <a style={margin10} className="btn btn-secondary" href={this.props.tel ? "tel:" + this.props.info : "mailto:" + this.props.info} target="_blank" rel="noopener noreferrer">{this.props.info}</a>
    );
  }
}

export default Contact;
