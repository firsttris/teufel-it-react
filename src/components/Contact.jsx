import React, { Component } from 'react';
import Social from './Social.jsx';
import SectionHeader from './SectionHeader.jsx';

export default class Contact extends Component {
  render() {
    return (
      <div className="row py-3 text-center">
        <div className="col">
          <span className="text-uppercase" style={{ fontSize: '30px', letterSpacing: '10px' }}>
            <SectionHeader title="CONTACT_TITLE" />
          </span>
          <div>
            <Social icon="fa-phone" url="tel:+49 176 45744166" />
            <Social icon="fa-envelope-o" url="mailto:info@teufel-it.de" />
            <Social icon="fa-xing-square" url="https://www.xing.com/profile/Tristan_Teufel" />
            <Social
              icon="fa-linkedin-square"
              url="https://de.linkedin.com/pub/tristan-teufel/8a/30a/b5"
            />
            <Social icon="fa-github" url="https://github.com/firsttris" />
          </div>
        </div>
      </div>
    );
  }
}
