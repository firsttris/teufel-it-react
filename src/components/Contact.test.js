import React from 'react';
import Contact from './Contact';
import renderer from 'react-test-renderer';

test('Display Contact', () => {
  const component = renderer.create(
    <Contact info="+49 176 45744166" tel={true}></Contact>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});