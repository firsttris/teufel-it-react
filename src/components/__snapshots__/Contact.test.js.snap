// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Contact 1`] = `
<a
  className="btn btn-secondary"
  href="tel:+49 176 45744166"
  rel="noopener noreferrer"
  style={
    Object {
      "margin": "10px",
    }
  }
  target="_blank"
>
  +49 176 45744166
</a>
`;
